### DISCUSSION TITLE
Ocelot Authorization using Keycloak identity management tool

### DISCUSSION THREAD

**Discussion by satishviswanathan** [ThreeMammals/Ocelot#1932] [Ideas]
Do you have any samples implementing authorization using Keyclock and  then forward the request to downstream services. 

### Update by Maintainer on Jan 20, 2024
Not  `Keyclock`, but `Keycloak` :point_up: :smile_cat:

**Comment by raman-m** [comment]
**No!** Our [samples](https://github.com/ThreeMammals/Ocelot/tree/develop/samples) are waiting for your contribution! :wink: 

I've just checked [Keycloak repo](https://github.com/keycloak/keycloak) and [Keycloak product](https://www.keycloak.org/) during this work week: this is amazing open-source project for Identity management! And, the source code is written in Java, and... seems there are no .NET [Client Adapters](https://www.keycloak.org/downloads) for this product (only JavaScript one). So, it will be hard to integrate to [Keycloak Rest API](https://www.keycloak.org/docs-api/23.0.4/rest-api/index.html)...

@ggnaegi FYI

**Reply by raman-m** [reply]
@RaynaldM Are you interested in Keycloak integration for your projects with Ocelot?

**Reply by RaynaldM** [reply]
no

**Reply by raman-m** [reply]
@RaynaldM It's OK. 
This week I've checked our lovely [Identity Server 4](https://identityserver4.readthedocs.io/en/latest/index.html) docs. It seems new Duende IdentityServer is free for open-source (I hope) and commercial use 👇 
- See [Warning](https://identityserver4.readthedocs.io/en/latest/index.html)
